test/svg-sprite/shape/shape.svg.test.js

```diff
@@ -255,3 +255,121 @@ describe('testing setSVG()', () => {
+        jest.spyOn(shape, 'getViewbox').mockReturnValueOnce(null);
+        jest.spyOn(shape, 'getDimensions').mockReturnValueOnce({ height: TEST_HEIGHT, width: TEST_WIDTH });
+        shape._setDimensions(jest.fn());
+
+        expect(shape.dom.documentElement.setAttribute).toHaveBeenCalledTimes(2);
+        expect(shape.dom.documentElement.setAttribute.mock.calls[0]).toStrictEqual(expect.arrayContaining(['height', TEST_HEIGHT]));
+        expect(shape.dom.documentElement.setAttribute.mock.calls[1]).toStrictEqual(expect.arrayContaining(['width', TEST_WIDTH]));
+    });
+
+    describe('shape need to be scaled', () => {
+        it('should scale if width is more than maxWidth', () => {
+            expect.hasAssertions();
+
+            const TEST_WIDTH = 400;
+            const TEST_HEIGHT = 100;
+            const shape = new SVGShape(TEST_FILE, TEST_SPRITER);
+            shape.width = TEST_WIDTH;
+            shape.height = TEST_HEIGHT;
+            shape.config.spacing.box = 'padding';
+            shape.config.spacing.padding.right = 10;
+            shape.config.spacing.padding.left = 10;
+            shape.config.dimension.maxWidth = TEST_WIDTH;
+
+            shape._setDimensions(jest.fn());
+
+            expect(shape.width).toBe(380);
+            expect(shape.height).toBe(95); // 100 * 280 / 300
+            expect(shape._scale).toBe(0.95); // 100 * 280 / 300
+        });
+
+        it('should scale if height is more than maxHeight', () => {
+            expect.hasAssertions();
+
+            const TEST_WIDTH = 300;
+            const TEST_HEIGHT = 100;
+            const shape = new SVGShape(TEST_FILE, TEST_SPRITER);
+            shape.width = TEST_WIDTH;
+            shape.height = TEST_HEIGHT;
+            shape.config.spacing.box = 'padding';
+            shape.config.spacing.padding.top = 10;
+            shape.config.spacing.padding.bottom = 10;
+            shape.config.dimension.maxHeight = TEST_HEIGHT;
+
+            shape._setDimensions(jest.fn());
+
+            expect(shape.height).toBe(80);
+            expect(shape.width).toBe(240); // 300 * 80 / 100
+            expect(shape._scale).toBe(0.8);
+        });
+
+        it('should scale if height is less than maxHeight and width is less than maxWidth nad spacing is icon', () => {
+            expect.hasAssertions();
+
+            const TEST_WIDTH = 300;
+            const TEST_HEIGHT = 100;
+            const shape = new SVGShape(TEST_FILE, TEST_SPRITER);
+            shape.width = TEST_WIDTH / 2;
+            shape.height = TEST_HEIGHT / 2;
+            shape.config.spacing.box = 'icon';
+            shape.config.dimension.maxHeight = TEST_HEIGHT;
+            shape.config.dimension.maxWidth = TEST_WIDTH;
+
+            shape._setDimensions(jest.fn());
+
+            expect(shape._scale).toBe(2);
+            expect(shape.height).toBe(100);
+            expect(shape.width).toBe(300); // 300 * 80 / 100
```

Comment:
the comment seems wrong here I think

Reply:
What do you think about using the actual expressions instead of the calculated result?

Reply:
I originally wrote here the expressions. But rewrote to calculated results. I think this might be OK, we should test it

Reply:
@XhmikosR it's much better now, thanks